apps/web/src/components/Auth/RegisterForm.tsx

```diff
@@ -0,0 +1,205 @@
+import { Form, Controller } from "@litespace/ui/Form";
+import { Button } from "@litespace/ui/Button";
+import { useToast } from "@litespace/ui/Toast";
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import React, { useCallback, useEffect, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { Route } from "@/types/routes";
+import { IUser } from "@litespace/types";
+import { useRegisterUser } from "@litespace/headless/user";
+import { useUserContext } from "@litespace/headless/context/user";
+import { Typography } from "@litespace/ui/Typography";
+import {
+  useValidateEmail,
+  useValidatePassword,
+} from "@litespace/ui/hooks/validation";
+import { useGoogle } from "@/hooks/google";
+import Google from "@litespace/assets/Google";
+import { getErrorMessageId } from "@litespace/ui/errorMessage";
+interface IForm {
+  email: string;
+  password: string;
+  confirmedPassword: string;
+}
+
+type Role = (typeof roles)[number];
+
+const roles = [IUser.Role.Tutor, IUser.Role.Student] as const;
+
+const origin = location.origin;
+const callbackUrl = origin.concat(Route.VerifyEmail);
```

Comment:
We can move the callback url defiend here https://github.com/litespace-org/litespace/blob/51205b4a3698bc35e527b049c94d6bfb1f7f4aeb/apps/web/src/types/routes.ts#L33-L35